### PR TITLE
Adding the Routing Indicator padding as described in the 3GPP TS 24.5…

### PR DESCRIPTION
**Pull request by Stg22**
When the Routing Indicator is less than 4 digits, the value needs to be completed according to the table 9.11.3.4.1 in section 9.11.3.4 from the document 3GPP TS 24.501.

This patch proposes  a quick adaptation of encodeBCD.

### CODE REVIEW

File: src/lib/nas/base.cpp [hunk 1]
```diff
@@ -45,13 +45,35 @@ void nas::EncodeBcdString(OctetString &stream, const std::string &bcd, size_t oc
     for (size_t i = 0; i < spare; i++)
         halfOctets[i + bcd.length() + (skipFirst ? 1 : 0)] = 0xF;
 
+    size_t octectCount = 0;
```
Comment (louisroyer) [inline src/lib/nas/base.cpp:48]:
```suggestion
    size_t octetCount = 0;
```

File: src/lib/nas/base.cpp [hunk 2]
```diff
@@ -45,13 +45,35 @@ void nas::EncodeBcdString(OctetString &stream, const std::string &bcd, size_t oc
...
     for (size_t i = 0; i < requiredHalfOctets / 2; i++)
     {
         int little = halfOctets[2 * i];
         int big = halfOctets[2 * i + 1];
         int octet = big << 4 | little;
         stream.appendOctet(octet);
+        octectCount++;
```
Comment (louisroyer) [inline src/lib/nas/base.cpp:55]:
```suggestion
        octetCount++;
```

File: src/lib/nas/base.cpp [hunk 3]
```diff
@@ -45,13 +45,35 @@ void nas::EncodeBcdString(OctetString &stream, const std::string &bcd, size_t oc
...
+    // (see NOTE 2). If no Routing Indicator is configured in the USIM, the
+    // UE shall code bits 1 to 4 of octet 8 of the Routing Indicator 
+    // as "0000" and the remaining digits as “1111".
+    //
+    // NOTE 2:	For a 3-digit Routing Indicator, e.g "567", bits 1 to 4 of
+    // octet 8 are coded as "0101", bits 5 to 8 of octet 8 are coded as
+    // "0110", bits 1 to 4 of octet 9 are coded as "0111", bits 5 to 8 of
+    // octet 9 are coded as "1111".
+    if (isRoutingIndicator) {
+        while (octectCount < octetLength) {
```
Comment (louisroyer) [inline src/lib/nas/base.cpp:72]:
```suggestion
        while (octetCount < octetLength) {
```

File: src/lib/nas/base.cpp [hunk 4]
```diff
@@ -45,13 +45,35 @@ void nas::EncodeBcdString(OctetString &stream, const std::string &bcd, size_t oc
...
+            stream.appendOctet(0xFF);
+            octectCount++;
```
Comment (louisroyer) [inline src/lib/nas/base.cpp:74]:
```suggestion
            octetCount++;
```

**Review by louisroyer — CHANGES_REQUESTED** [review]
Typo in variable name: it should be `octetCount` and not `octectCount`.

**Comment by Stg22** [comment]
Sorry for the tipo on only so few modificated code :o(

**Comment by aligungr** [comment]
Modifying `EncodeBcdString` function and accepting another parameter `bool isRoutingIndicator` would be confusing.
It's better to write a separate function to encode Routing Indicator, such as `EncodeRoutingIndicator`.
Could you fix this?

Thank you.

**Comment by Stg22** [comment]
Is it acceptable to modify EncodeBcdString to return the number of bytes pushed into the stream parameter? It will return size_t instead of void and the function EncodeRoutingIndicator could use EncodeBbdSting in easy way?

**Pull request closed without merging**